### PR TITLE
Transformer - Update

### PR DESCRIPTION
**Pull request by DyeKuu**
**What this pull request does**:
- moves all neural network settings into `config.py` as well as the build model function, transformer block.
- `cross_validation.py` tries to split the training data into k-fold based on the day. (i.e day1-20 for training, day20-30 for validation. Then 20-30 for training, 1-20 for validation.) We set k models to train to see if there is any overfitting. All is well.
- `data/AMF_test_Y.csv` is all the labels we assume to be known.
- `data_augment.py` does the data augmentation: adds new columns to train & test. The resulting files are `data/Agmt_train_X.csv` & `data/Agmt_test_X.csv`.
- `dl.py` does all the stuff in keras api. This is where 0.97 comes from!
- An idea is to do the pretraining of **predict next line** on both train & test set, then do the fine-tuning of classification on only train set. The implementation is in `pretraining.py` and `finetuning.py`. the result is just 0.9
- `predict.py` loads the model_weights then does the prediction.
- `produce_data.py` combines the train set and some known traders in test set. Run it and you will get `data/Agmtd_train_X.csv` & `data/Agmtd_train_Y.csv`, which are too big to be put here.
- many other results.

### CODE REVIEW

**Review by DyeKuu — COMMENTED** [review]
👍 Looks good to me
